File: core/save_slots.py
"""
Save Slots System - Multiple save file management.
Allows players to have multiple save files with different ducks.
"""
from dataclasses import dataclass, field
from datetime import datetime
from typing import Dict, List, Optional, Tuple
from pathlib import Path
import json
import os
import shutil


@dataclass
class SaveSlotInfo:
    """Information about a save slot."""
    slot_id: int
    is_empty: bool = True
    duck_name: str = ""
    level: int = 1
    playtime_minutes: int = 0
    coins: int = 0
    created_at: str = ""
    last_played: str = ""
    prestige_level: int = 0
    achievements_count: int = 0
    mood: str = "happy"
    preview_ascii: List[str] = field(default_factory=list)


class SaveSlotsSystem:
    """
    System for managing multiple save slots.
    """
    
    MAX_SLOTS = 5
    
    def __init__(self, save_dir: str = "~/.cheese_the_duck"):
        self.save_dir = Path(save_dir).expanduser()
        self.save_dir.mkdir(parents=True, exist_ok=True)
        
        self.current_slot: int = 1
        self.slots: Dict[int, SaveSlotInfo] = {}
        
        # Initialize slot info
        self.refresh_slots()
    
    def get_save_path(self, slot_id: int) -> Path:
        """Get the save file path for a slot."""
        if slot_id == 1:
            return self.save_dir / "save.json"  # Legacy default
        return self.save_dir / f"save_slot_{slot_id}.json"
    
    def get_backup_path(self, slot_id: int) -> Path:
        """Get the backup file path for a slot."""
        return self.save_dir / f"save_slot_{slot_id}.backup.json"
    
    def refresh_slots(self):
        """Refresh information about all save slots."""
        self.slots = {}
        
        for slot_id in range(1, self.MAX_SLOTS + 1):
            save_path = self.get_save_path(slot_id)
            
            if save_path.exists():
                try:
                    with open(save_path, 'r') as f:
                        data = json.load(f)
                    
                    self.slots[slot_id] = self._parse_save_data(slot_id, data)
                except (json.JSONDecodeError, KeyError, IOError):
                    self.slots[slot_id] = SaveSlotInfo(
                        slot_id=slot_id,
                        is_empty=False,
                        duck_name="CORRUPTED",
                    )
            else:
                self.slots[slot_id] = SaveSlotInfo(
                    slot_id=slot_id,
                    is_empty=True,
                )
    
    def _parse_save_data(self, slot_id: int, data: dict) -> SaveSlotInfo:
        """Parse save data into slot info."""
        duck_data = data.get("duck", {})
        progression_data = data.get("progression", {})
        prestige_data = data.get("prestige", {})
        achievements_data = data.get("achievements", {})
        stats_data = data.get("statistics", {})
        
        # Get duck mood for preview
        mood_data = duck_data.get("mood", {})
        current_mood = mood_data.get("current_mood", "happy")
        
        # Create preview ASCII based on level/prestige
        level = progression_data.get("level", 1)
        prestige = prestige_data.get("prestige_level", 0)
        
        preview = self._generate_preview(level, prestige, current_mood)
        
        return SaveSlotInfo(
            slot_id=slot_id,
            is_empty=False,
            duck_name=duck_data.get("name", "Cheese"),
            level=level,
            playtime_minutes=stats_data.get("total_playtime_minutes", 0),
            coins=progression_data.get("coins", 0),
            created_at=stats_data.get("first_played", "Unknown"),
            last_played=stats_data.get("last_played", "Unknown"),
            prestige_level=prestige,
            achievements_count=len(achievements_data.get("unlocked", [])),
            mood=current_mood,
            preview_ascii=preview,
        )
    
    def _generate_preview(self, level: int, prestige: int, mood: str) -> List[str]:
        """Generate ASCII preview for a save slot."""
        # Mood faces
        mood_faces = {
            "happy": "^◡^",
            "excited": "*◡*",
            "content": "◡‿◡",
            "sad": "╥﹏╥",
            "hungry": "ò﹏ó",
            "sleepy": "-_-",
            "playful": "◕‿◕",
        }
        face = mood_faces.get(mood, "^◡^")
        
        # Base duck
        if prestige > 0:
            # Prestige duck with crown/aura
            return [
                f"  *^*   ",
                f"   ({face})  ",
                f"  >d<    ",
                f"  P{prestige} Lv{level} ",
            ]
        elif level >= 50:
            return [
                f"    *     ",
                f"   ({face})  ",
                f"   d      ",
                f"   Lv{level}   ",
            ]
        else:
            return [
                f"   ({face})  ",
                f"   d      ",
                f"   Lv{level}   ",
            ]
    
    def format_playtime(self, minutes: int) -> str:
        """Format playtime as human-readable string."""
        hours = minutes // 60
        mins = minutes % 60
        
        if hours > 24:
            days = hours // 24
            hours = hours % 24
            return f"{days}d {hours}h"
        elif hours > 0:
            return f"{hours}h {mins}m"
        else:
            return f"{mins}m"
    
    def get_slot(self, slot_id: int) -> Optional[SaveSlotInfo]:
        """Get info for a specific slot."""
        return self.slots.get(slot_id)
    
    def load_slot(self, slot_id: int) -> Optional[dict]:
        """Load save data from a slot."""
        if slot_id not in range(1, self.MAX_SLOTS + 1):
            return None
        
        save_path = self.get_save_path(slot_id)
        if not save_path.exists():
            return None
        
        try:
            with open(save_path, 'r') as f:
                return json.load(f)
        except (json.JSONDecodeError, IOError):
            return None
    
    def save_to_slot(self, slot_id: int, data: dict) -> bool:
        """Save data to a slot."""
        if slot_id not in range(1, self.MAX_SLOTS + 1):
            return False
        
        save_path = self.get_save_path(slot_id)
        backup_path = self.get_backup_path(slot_id)
        
        try:
            # Create backup of existing save
            if save_path.exists():
                shutil.copy2(save_path, backup_path)
            
            # Save new data
            with open(save_path, 'w') as f:
                json.dump(data, f, indent=2)
            
            # Refresh slot info
            self.refresh_slots()
            return True
            
        except IOError:
            return False
    
    def delete_slot(self, slot_id: int) -> bool:
        """Delete a save slot."""
        if slot_id not in range(1, self.MAX_SLOTS + 1):
            return False
        
        save_path = self.get_save_path(slot_id)
        backup_path = self.get_backup_path(slot_id)
        
        try:
            if save_path.exists():
                save_path.unlink()
            if backup_path.exists():
                backup_path.unlink()
            
            self.refresh_slots()
            return True
            
        except IOError:
            return False
    
    def copy_slot(self, from_slot: int, to_slot: int) -> bool:
        """Copy save from one slot to another."""
        data = self.load_slot(from_slot)
        if data is None:
            return False
        
        return self.save_to_slot(to_slot, data)
    
    def restore_backup(self, slot_id: int) -> bool:
        """Restore a slot from its backup."""
        backup_path = self.get_backup_path(slot_id)
        save_path = self.get_save_path(slot_id)
        
        if not backup_path.exists():
            return False
        
        try:
            shutil.copy2(backup_path, save_path)
            self.refresh_slots()
            return True
        except IOError:
            return False
    
    def has_backup(self, slot_id: int) -> bool:
        """Check if a slot has a backup file."""
        return self.get_backup_path(slot_id).exists()
    
    def export_slot(self, slot_id: int, export_path: str) -> bool:
        """Export a save slot to an external file."""
        data = self.load_slot(slot_id)
        if data is None:
            return False
        
        try:
            with open(export_path, 'w') as f:
                json.dump(data, f, indent=2)
            return True
        except IOError:
            return False
    
    def import_slot(self, slot_id: int, import_path: str) -> bool:
        """Import a save file into a slot."""
        try:
            with open(import_path, 'r') as f:
                data = json.load(f)
            return self.save_to_slot(slot_id, data)
        except (json.JSONDecodeError, IOError):
            return False
    
    def render_slot_selection(self, show_details: bool = True) -> List[str]:
        """Render the save slot selection screen."""
        lines = [
            "+===============================================+",
            "|           [=] SAVE SLOTS [=]                    |",
            "+===============================================+",
        ]
        
        for slot_id in range(1, self.MAX_SLOTS + 1):
            slot = self.slots.get(slot_id)
            
            if slot is None or slot.is_empty:
                lines.append(f"|  [{slot_id}] --- EMPTY SLOT ---                   |")
                lines.append("|      Start a new adventure!                   |")
                lines.append("|                                               |")
            else:
                # Active indicator
                active = " <" if slot_id == self.current_slot else "  "
                
                lines.append(f"|  [{slot_id}] {slot.duck_name:<25} {active}      |")
                
                if show_details:
                    # Show preview ASCII
                    for preview_line in slot.preview_ascii:
                        lines.append(f"|      {preview_line:<39}  |")
                    
                    # Stats
                    playtime = self.format_playtime(slot.playtime_minutes)
                    lines.append(f"|      $ {slot.coins:<8}  ⏱️ {playtime:<15}  |")
                    lines.append(f"|      [#] {slot.achievements_count} achievements                    |")
                    
                    # Last played
                    if slot.last_played and slot.last_played != "Unknown":
                        try:
                            dt = datetime.fromisoformat(slot.last_played)
                            last_str = dt.strftime("%Y-%m-%d %H:%M")
                        except ValueError:
                            last_str = slot.last_played[:16]
                        lines.append(f"|      Last: {last_str:<30}  |")
                
                lines.append("|                                               |")
        
        lines.extend([
            "+===============================================+",
            "|  [1-5] Select  [D] Delete  [C] Copy           |",
            "|  [E] Export    [I] Import  [B] Back           |",
            "+===============================================+",
        ])
        
        return lines
    
    def render_slot_details(self, slot_id: int) -> List[str]:
        """Render detailed view of a single slot."""
        slot = self.slots.get(slot_id)
        
        lines = [
            "+===============================================+",
            f"|           [=] SLOT {slot_id} DETAILS                   |",
            "+===============================================+",
        ]
        
        if slot is None or slot.is_empty:
            lines.append("|                                               |")
            lines.append("|              Empty Save Slot                  |")
            lines.append("|                                               |")
            lines.append("|         Start a new adventure?                |")
            lines.append("|                                               |")
        else:
            # Duck preview centered
            lines.append("|                                               |")
            for preview_line in slot.preview_ascii:
                lines.append(f"|  {preview_line:^43}  |")
            
            lines.append("|                                               |")
            lines.append(f"|  Duck Name: {slot.duck_name:<32}  |")
            lines.append(f"|  Level: {slot.level:<36}  |")
            
            if slot.prestige_level > 0:
                lines.append(f"|  Prestige: {slot.prestige_level:<33}  |")
            
            lines.append(f"|  Coins: {slot.coins:<36}  |")
            lines.append(f"|  Playtime: {self.format_playtime(slot.playtime_minutes):<33}  |")
            lines.append(f"|  Achievements: {slot.achievements_count:<28}  |")
            lines.append(f"|  Current Mood: {slot.mood:<28}  |")
            
            lines.append("|                                               |")
            
            if slot.created_at and slot.created_at != "Unknown":
                try:
                    dt = datetime.fromisoformat(slot.created_at)
                    created = dt.strftime("%Y-%m-%d")
                except ValueError:
                    created = slot.created_at[:10]
                lines.append(f"|  Created: {created:<34}  |")
            
            if slot.last_played and slot.last_played != "Unknown":
                try:
                    dt = datetime.fromisoformat(slot.last_played)
                    last = dt.strftime("%Y-%m-%d %H:%M")
                except ValueError:
                    last = slot.last_played[:16]
                lines.append(f"|  Last Played: {last:<30}  |")
            
            has_backup = self.has_backup(slot_id)
            backup_str = "Yes" if has_backup else "No"
            lines.append(f"|  Backup Available: {backup_str:<25}  |")
        
        lines.extend([
            "+===============================================+",
            "|  [L] Load  [D] Delete  [R] Restore Backup     |",
            "|  [E] Export  [B] Back                         |",
            "+===============================================+",
        ])
        
        return lines
    
    def to_dict(self) -> dict:
        """Convert to dictionary for meta-save (which slot was last used)."""
        return {
            "current_slot": self.current_slot,
        }
    
    @classmethod
    def from_dict(cls, data: dict, save_dir: str = "~/.cheese_the_duck") -> "SaveSlotsSystem":
        """Create from dictionary."""
        system = cls(save_dir)
        system.current_slot = data.get("current_slot", 1)
        return system


# Global instance
save_slots_system = SaveSlotsSystem()

File: StupidDuckCS/src/StupidDuck.Core/SaveSlots.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text.Json;

namespace StupidDuck.Core;

/// <summary>
/// Information about a save slot.
/// </summary>
public class SaveSlotInfo
{
    public int SlotId { get; set; }
    public bool IsEmpty { get; set; } = true;
    public string DuckName { get; set; } = "";
    public int Level { get; set; } = 1;
    public int PlaytimeMinutes { get; set; }
    public int Coins { get; set; }
    public string CreatedAt { get; set; } = "";
    public string LastPlayed { get; set; } = "";
    public int PrestigeLevel { get; set; }
    public int AchievementsCount { get; set; }
    public string Mood { get; set; } = "happy";
    public List<string> PreviewAscii { get; set; } = new();
}

/// <summary>
/// System for managing multiple save slots.
/// </summary>
public class SaveSlotsSystem
{
    public const int MaxSlots = 5;

    public string SaveDir { get; }
    public int CurrentSlot { get; set; } = 1;
    public Dictionary<int, SaveSlotInfo> Slots { get; set; } = new();

    private static readonly Dictionary<string, string> MoodFaces = new()
    {
        ["happy"] = "^◡^",
        ["excited"] = "✧◡✧",
        ["content"] = "◡‿◡",
        ["sad"] = "╥﹏╥",
        ["hungry"] = "ò﹏ó",
        ["sleepy"] = "-_-",
        ["playful"] = "◕‿◕"
    };

    public SaveSlotsSystem(string? saveDir = null)
    {
        SaveDir = saveDir ?? Path.Combine(
            Environment.GetFolderPath(Environment.SpecialFolder.UserProfile),
            ".cheese_the_duck");

        Directory.CreateDirectory(SaveDir);
        RefreshSlots();
    }

    public string GetSavePath(int slotId) =>
        slotId == 1
            ? Path.Combine(SaveDir, "save.json")
            : Path.Combine(SaveDir, $"save_slot_{slotId}.json");

    public string GetBackupPath(int slotId) =>
        Path.Combine(SaveDir, $"save_slot_{slotId}.backup.json");

    public void RefreshSlots()
    {
        Slots.Clear();

        for (var slotId = 1; slotId <= MaxSlots; slotId++)
        {
            var savePath = GetSavePath(slotId);

            if (File.Exists(savePath))
            {
                try
                {
                    var json = File.ReadAllText(savePath);
                    using var doc = JsonDocument.Parse(json);
                    var data = doc.RootElement;
                    Slots[slotId] = ParseSaveData(slotId, data);
                }
                catch
                {
                    Slots[slotId] = new SaveSlotInfo
                    {
                        SlotId = slotId,
                        IsEmpty = false,
                        DuckName = "CORRUPTED"
                    };
                }
            }
            else
            {
                Slots[slotId] = new SaveSlotInfo
                {
                    SlotId = slotId,
                    IsEmpty = true
                };
            }
        }
    }

    private SaveSlotInfo ParseSaveData(int slotId, JsonElement data)
    {
        var duckData = data.TryGetProperty("duck", out var d) ? d : default;
        var progressionData = data.TryGetProperty("progression", out var p) ? p : default;
        var prestigeData = data.TryGetProperty("prestige", out var pr) ? pr : default;
        var achievementsData = data.TryGetProperty("achievements", out var a) ? a : default;
        var statsData = data.TryGetProperty("statistics", out var s) ? s : default;

        var duckName = TryGetString(duckData, "name") ?? "Cheese";
        var level = TryGetInt(progressionData, "level") ?? 1;
        var coins = TryGetInt(progressionData, "coins") ?? 0;
        var prestigeLevel = TryGetInt(prestigeData, "prestige_level") ?? 0;
        var playtime = TryGetInt(statsData, "total_playtime_minutes") ?? 0;
        var firstPlayed = TryGetString(statsData, "first_played") ?? "Unknown";
        var lastPlayed = TryGetString(statsData, "last_played") ?? "Unknown";

        // Get mood
        var moodData = duckData.ValueKind == JsonValueKind.Object && duckData.TryGetProperty("mood", out var m) ? m : default;
        var currentMood = TryGetString(moodData, "current_mood") ?? "happy";

        // Count achievements
        var achievementsCount = 0;
        if (achievementsData.ValueKind == JsonValueKind.Object &&
            achievementsData.TryGetProperty("unlocked", out var unlocked) &&
            unlocked.ValueKind == JsonValueKind.Array)
        {
            achievementsCount = unlocked.GetArrayLength();
        }

        var preview = GeneratePreview(level, prestigeLevel, currentMood);

        return new SaveSlotInfo
        {
            SlotId = slotId,
            IsEmpty = false,
            DuckName = duckName,
            Level = level,
            PlaytimeMinutes = playtime,
            Coins = coins,
            CreatedAt = firstPlayed,
            LastPlayed = lastPlayed,
            PrestigeLevel = prestigeLevel,
            AchievementsCount = achievementsCount,
            Mood = currentMood,
            PreviewAscii = preview
        };
    }

    private static string? TryGetString(JsonElement element, string property)
    {
        if (element.ValueKind == JsonValueKind.Object &&
            element.TryGetProperty(property, out var val) &&
            val.ValueKind == JsonValueKind.String)
            return val.GetString();
        return null;
    }

    private static int? TryGetInt(JsonElement element, string property)
    {
        if (element.ValueKind == JsonValueKind.Object &&
            element.TryGetProperty(property, out var val) &&
            val.ValueKind == JsonValueKind.Number)
            return val.GetInt32();
        return null;
    }

    private List<string> GeneratePreview(int level, int prestige, string mood)
    {
        var face = MoodFaces.GetValueOrDefault(mood, "^◡^");

        if (prestige > 0)
        {
            return new List<string>
            {
                "  ✨👑✨   ",
                $"   ({face})  ",
                "  >🦆<    ",
                $"  P{prestige} Lv{level} "
            };
        }
        else if (level >= 50)
        {
            return new List<string>
            {
                "    ⭐     ",
                $"   ({face})  ",
                "   🦆      ",
                $"   Lv{level}   "
            };
        }
        else
        {
            return new List<string>
            {
                $"   ({face})  ",
                "   🦆      ",
                $"   Lv{level}   "
            };
        }
    }

    public string FormatPlaytime(int minutes)
    {
        var hours = minutes / 60;
        var mins = minutes % 60;

        if (hours > 24)
        {
            var days = hours / 24;
            hours %= 24;
            return $"{days}d {hours}h";
        }
        else if (hours > 0)
        {
            return $"{hours}h {mins}m";
        }
        else
        {
            return $"{mins}m";
        }
    }

    public SaveSlotInfo? GetSlot(int slotId) =>
        Slots.GetValueOrDefault(slotId);

    public string? LoadSlot(int slotId)
    {
        if (slotId < 1 || slotId > MaxSlots)
            return null;

        var savePath = GetSavePath(slotId);
        if (!File.Exists(savePath))
            return null;

        try
        {
            return File.ReadAllText(savePath);
        }
        catch
        {
            return null;
        }
    }

    public bool SaveToSlot(int slotId, string jsonData)
    {
        if (slotId < 1 || slotId > MaxSlots)
            return false;

        var savePath = GetSavePath(slotId);
        var backupPath = GetBackupPath(slotId);

        try
        {
            // Create backup of existing save
            if (File.Exists(savePath))
                File.Copy(savePath, backupPath, overwrite: true);

            // Save new data
            File.WriteAllText(savePath, jsonData);

            RefreshSlots();
            return true;
        }
        catch
        {
            return false;
        }
    }

    public bool DeleteSlot(int slotId)
    {
        if (slotId < 1 || slotId > MaxSlots)
            return false;

        var savePath = GetSavePath(slotId);
        var backupPath = GetBackupPath(slotId);

        try
        {
            if (File.Exists(savePath))
                File.Delete(savePath);
            if (File.Exists(backupPath))
                File.Delete(backupPath);

            RefreshSlots();
            return true;
        }
        catch
        {
            return false;
        }
    }

    public bool CopySlot(int fromSlot, int toSlot)
    {
        var data = LoadSlot(fromSlot);
        if (data == null)
            return false;

        return SaveToSlot(toSlot, data);
    }

    public bool RestoreBackup(int slotId)
    {
        var backupPath = GetBackupPath(slotId);
        var savePath = GetSavePath(slotId);

        if (!File.Exists(backupPath))
            return false;

        try
        {
            File.Copy(backupPath, savePath, overwrite: true);
            RefreshSlots();
            return true;
        }
        catch
        {
            return false;
        }
    }

    public bool HasBackup(int slotId) =>
        File.Exists(GetBackupPath(slotId));

    public bool ExportSlot(int slotId, string exportPath)
    {
        var data = LoadSlot(slotId);
        if (data == null)
            return false;

        try
        {
            File.WriteAllText(exportPath, data);
            return true;
        }
        catch
        {
            return false;
        }
    }

    public bool ImportSlot(int slotId, string importPath)
    {
        try
        {
            var data = File.ReadAllText(importPath);
            // Validate JSON
            using var _ = JsonDocument.Parse(data);
            return SaveToSlot(slotId, data);
        }
        catch
        {
            return false;
        }
    }

    public List<string> RenderSlotSelection(bool showDetails = true)
    {
        var lines = new List<string>
        {
            "╔═══════════════════════════════════════════════╗",
            "║           💾 SAVE SLOTS 💾                    ║",
            "╠═══════════════════════════════════════════════╣"
        };

        for (var slotId = 1; slotId <= MaxSlots; slotId++)
        {
            var slot = Slots.GetValueOrDefault(slotId);

            if (slot == null || slot.IsEmpty)
            {
                lines.Add($"║  [{slotId}] ─── EMPTY SLOT ───                   ║");
                lines.Add("║      Start a new adventure!                   ║");
                lines.Add("║                                               ║");
            }
            else
            {
                var active = slotId == CurrentSlot ? " ◀" : "  ";
                var name = slot.DuckName.PadRight(25);
                if (name.Length > 25) name = name.Substring(0, 25);
                lines.Add($"║  [{slotId}] {name} {active}      ║");

                if (showDetails)
                {
                    foreach (var previewLine in slot.PreviewAscii)
                    {
                        var padded = previewLine.PadRight(39);
                        if (padded.Length > 39) padded = padded.Substring(0, 39);
                        lines.Add($"║      {padded}  ║");
                    }

                    var playtime = FormatPlaytime(slot.PlaytimeMinutes);
                    lines.Add($"║      💰 {slot.Coins,-8}  ⏱️ {playtime,-15}  ║");
                    lines.Add($"║      🏆 {slot.AchievementsCount} achievements                    ║");

                    if (!string.IsNullOrEmpty(slot.LastPlayed) && slot.LastPlayed != "Unknown")
                    {
                        var lastStr = slot.LastPlayed.Length > 16 ? slot.LastPlayed.Substring(0, 16) : slot.LastPlayed;
                        if (DateTime.TryParse(slot.LastPlayed, out var dt))
                            lastStr = dt.ToString("yyyy-MM-dd HH:mm");
                        lines.Add($"║      Last: {lastStr.PadRight(30)}  ║");
                    }
                }

                lines.Add("║                                               ║");
            }
        }

        lines.Add("╠═══════════════════════════════════════════════╣");
        lines.Add("║  [1-5] Select  [D] Delete  [C] Copy           ║");
        lines.Add("║  [E] Export    [I] Import  [B] Back           ║");
        lines.Add("╚═══════════════════════════════════════════════╝");

        return lines;
    }

    public List<string> RenderSlotDetails(int slotId)
    {
        var slot = Slots.GetValueOrDefault(slotId);

        var lines = new List<string>
        {
            "╔═══════════════════════════════════════════════╗",
            $"║           💾 SLOT {slotId} DETAILS                   ║",
            "╠═══════════════════════════════════════════════╣"
        };

        if (slot == null || slot.IsEmpty)
        {
            lines.Add("║                                               ║");
            lines.Add("║              Empty Save Slot                  ║");
            lines.Add("║                                               ║");
            lines.Add("║         Start a new adventure?                ║");
            lines.Add("║                                               ║");
        }
        else
        {
            lines.Add("║                                               ║");
            foreach (var previewLine in slot.PreviewAscii)
            {
                var centered = previewLine.PadLeft((43 + previewLine.Length) / 2).PadRight(43);
                if (centered.Length > 43) centered = centered.Substring(0, 43);
                lines.Add($"║  {centered}  ║");
            }

            lines.Add("║                                               ║");
            lines.Add($"║  Duck Name: {slot.DuckName.PadRight(32)}  ║");
            lines.Add($"║  Level: {slot.Level.ToString().PadRight(36)}  ║");

            if (slot.PrestigeLevel > 0)
                lines.Add($"║  Prestige: {slot.PrestigeLevel.ToString().PadRight(33)}  ║");

            lines.Add($"║  Coins: {slot.Coins.ToString().PadRight(36)}  ║");
            lines.Add($"║  Playtime: {FormatPlaytime(slot.PlaytimeMinutes).PadRight(33)}  ║");
            lines.Add($"║  Achievements: {slot.AchievementsCount.ToString().PadRight(28)}  ║");
            lines.Add($"║  Current Mood: {slot.Mood.PadRight(28)}  ║");

            lines.Add("║                                               ║");

            if (!string.IsNullOrEmpty(slot.CreatedAt) && slot.CreatedAt != "Unknown")
            {
                var created = slot.CreatedAt.Length > 10 ? slot.CreatedAt.Substring(0, 10) : slot.CreatedAt;
                if (DateTime.TryParse(slot.CreatedAt, out var dt))
                    created = dt.ToString("yyyy-MM-dd");
                lines.Add($"║  Created: {created.PadRight(34)}  ║");
            }

            if (!string.IsNullOrEmpty(slot.LastPlayed) && slot.LastPlayed != "Unknown")
            {
                var last = slot.LastPlayed.Length > 16 ? slot.LastPlayed.Substring(0, 16) : slot.LastPlayed;
                if (DateTime.TryParse(slot.LastPlayed, out var dt))
                    last = dt.ToString("yyyy-MM-dd HH:mm");
                lines.Add($"║  Last Played: {last.PadRight(30)}  ║");
            }

            var backupStr = HasBackup(slotId) ? "Yes" : "No";
            lines.Add($"║  Backup Available: {backupStr.PadRight(25)}  ║");
        }

        lines.Add("╠═══════════════════════════════════════════════╣");
        lines.Add("║  [L] Load  [D] Delete  [R] Restore Backup     ║");
        lines.Add("║  [E] Export  [B] Back                         ║");
        lines.Add("╚═══════════════════════════════════════════════╝");

        return lines;
    }

    // =============================================================================
    // SERIALIZATION
    // =============================================================================

    public SaveSlotsSaveData ToSaveData() => new()
    {
        CurrentSlot = CurrentSlot
    };

    public static SaveSlotsSystem FromSaveData(SaveSlotsSaveData data, string? saveDir = null)
    {
        var system = new SaveSlotsSystem(saveDir)
        {
            CurrentSlot = data.CurrentSlot > 0 ? data.CurrentSlot : 1
        };
        return system;
    }
}

public class SaveSlotsSaveData
{
    public int CurrentSlot { get; set; } = 1;
}
